### PR TITLE
adding dimensioned coagulation kernel

### LINKED ISSUE #58: complete coagulation calculation
- add dimensioned form
- add other options (theories)
- add tests for ion--particle and charged coagulation

### PR DESCRIPTION
**Pull request by ngam**
Fixes #58 

- dimensioned coagulation with options
- associated tests
- physics-based test (ion--particle -> 1e-12 in a specific case)

### CODE REVIEW

**Comment by ngam** [comment]
The physics-based test at the very end is legit, "EXP" (unpublished) in this figure is experimental and the rest are modeled values. 1e-12 m**3/s = 1e-6 cc**3/s (the preferred unit for such things)

[modelBetaValues.pdf](https://github.com/uncscode/particula/files/7742155/modelBetaValues.pdf)

File: particula/aerosol_dynamics/particle.py [hunk 1]
```diff
@@ -264,26 +264,124 @@ def collision_kernel_kinetic_limit(
...
+        Checks units: [dimensionless]
+
+        Paramaters:
+            self:           particle 1
+            other:          particle 2
+            environment:    environment conditions
+            authors:        authors of the parameterization
+                - gh2012    https://doi.org/10.1103/PhysRevE.78.046402
+                - cg2020    https://doi.org/XXXXXXXXXXXXXXXXXXXXXXXXXX
+                - hs        https://doi.org/XXXXXXXXXXXXXXXXXXXXXXXXXX
```
Comment (Gorkowski) [inline particula/aerosol_dynamics/particle.py:284]: add that hs: hard sphere approximation. 
it took a minute for me to figure out what that option was :)

Comment (Gorkowski) [reply particula/aerosol_dynamics/particle.py:284]: or call it hard_sphere

**Review by Gorkowski — CHANGES_REQUESTED** [review]
thats it, rest looks good to me

**Review by Gorkowski — APPROVED** [review]

**Comment by ngam** [comment]
Sorry, I disappeared, there was a little bug. I will push that quickly

**Pull request merged**